presto-main/src/main/java/com/facebook/presto/sql/relational/SqlFunctionUtils.java

```diff
@@ -108,6 +108,7 @@ public static RowExpression getSqlFunctionRowExpression(
                         Optional.empty(),
                         sqlFunctionProperties,
                         sessionFunctions,
+                        false,
```

Comment:
is this safe to assume it's not native?

Reply:
I don't see this code path being triggered in native path.
It is used in 
- https://github.com/vermapratyush/presto/blob/d5eed561c31e21cb0b6cfa5f94038d9d4f2b4b89/presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/InlineSqlFunctions.java#L106
https://github.com/prestodb/presto/blob/d5eed561c31e21cb0b6cfa5f94038d9d4f2b4b89/presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/InlineSqlFunctions.java#L106 - should not be invoked in native flow
- 
- https://github.com/vermapratyush/presto/blob/d5eed561c31e21cb0b6cfa5f94038d9d4f2b4b89/presto-main/src/main/java/com/facebook/presto/sql/gen/RowExpressionCompiler.java#L144
https://github.com/prestodb/presto/blob/d5eed561c31e21cb0b6cfa5f94038d9d4f2b4b89/presto-main/src/main/java/com/facebook/presto/sql/gen/RowExpressionCompiler.java#L144 - should not be invoked in native flow
- https://github.com/vermapratyush/presto/blob/d5eed561c31e21cb0b6cfa5f94038d9d4f2b4b89/presto-main/src/main/java/com/facebook/presto/sql/planner/RowExpressionInterpreter.java#L282
https://github.com/prestodb/presto/blob/d5eed561c31e21cb0b6cfa5f94038d9d4f2b4b89/presto-main/src/main/java/com/facebook/presto/sql/planner/RowExpressionInterpreter.java#L282 - Not sure about this one (i don’t see this getting invoked from the e2e unit tests)
Do you know if RowExpressionInterpreter is of concern?

Reply:
I think these can be invoked from native flow if someone uses a sql function that uses NULLIF in its implementation. And seems like an easy place for bugs to lurk.

Reply:
Could this be because all the native tests set configuration "inline-sql-functions" to false ? https://github.com/prestodb/presto/blob/master/presto-native-execution/src/test/java/com/facebook/presto/nativeworker/NativeQueryRunnerUtils.java#L32

Ideally, we should try to avoid such properties/configurations. 

But is it possible to use that property instead in this rule ?

Reply:
@aditi-pandit Can you please elaborate what do you mean by `can we use that property instead of this rule`?

Reply:
@vermapratyush : I'm not super familiar with optimizer code, but it is possible that instead of hardcoding to false, we get the value for inline-sql-functions property at a higher level (or here itself if there is access to it) and use that instead in this code ?

Reply:
The issue with `inline-sql-functions` is the same as that of using Session to determine if the query is for native, certain code-flow does not have session object.
I also verified with the velox team, and it would not make sense to provide support for SQL function for native queries within the java stack. It is better to support sql function by implementing it natively in velox itself. So I think hard-coding is ok here.